### PR TITLE
added sampler aggregation

### PR DESCRIPTION
**Pull request by einorler**
part of #97

### CODE REVIEW

File: src/Aggregation/SamplerAggregation.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     * {@inheritdoc}
+     */
+    public function getArray()
+    {
+        $out = array_filter(
+            [
+                'field' => $this->getField(),
+                'shard_size' => $this->getShardSize(),
+            ],
+            function ($val) {
```
Comment (saimaz) [inline src/Aggregation/SamplerAggregation.php]: This callback is not necessary.

File: tests/Aggregation/SamplerAggregationTest.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+
+namespace ONGR\ElasticsearchDSL\Tests\Aggregation;
+
+use ONGR\ElasticsearchDSL\Aggregation\SamplerAggregation;
+
+/**
+ * Unit test for children aggregation.
+ */
+class SamplerAggregationTest extends \PHPUnit_Framework_TestCase
```
Comment (saimaz) [inline tests/Aggregation/SamplerAggregationTest.php]: Add test case with name and field without shard size.

File: src/Aggregation/SamplerAggregation.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * Inner aggregations container init.
+     *
+     * @param string $name
+     * @param string $field
+     * @param int    $shardSize
+     */
+    public function __construct(
+        $name,
+        $field = null,
```
Comment (saimaz) [inline src/Aggregation/SamplerAggregation.php]: IMO field should be mandatory to set. No?

Comment (einorler) [inline src/Aggregation/SamplerAggregation.php]: No, the documantation states that the field is optional:
https://www.elastic.co/guide/en/elasticsearch/reference/2.3/search-aggregations-bucket-sampler-aggregation.html#_controlling_diversity

File: src/Aggregation/SamplerAggregation.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+     */
+    public function getArray()
+    {
+        $out = array_filter(
+            [
+                'field' => $this->getField(),
+                'shard_size' => $this->getShardSize(),
+            ]
+        );
+        return $this->processArray($out);
```
Comment (saimaz) [inline src/Aggregation/SamplerAggregation.php]: Array already is already processed in `AbstractAggregation:toArray()` https://github.com/ongr-io/ElasticsearchDSL/blob/master/src/Aggregation/AbstractAggregation.php#L125

**Comment by einorler** [comment]
Removed processing of parameters from `getArray` method

**Pull request merged**